src/runtime-rs/crates/agent/src/types.rs

```diff
@@ -390,6 +390,32 @@ pub struct KernelModule {
     pub parameters: Vec<String>,
 }
 
+impl KernelModule {
+    // input string: "ModuleName Param1 Param2 ..."
+    pub fn new(str: String) -> Self {
```

Comment:
What if `str == ""`? Is that a problem? If so, maybe we could `impl TryFrom<...> for KernelModule`?

Reply:
Thanks for reminding. This case is handled in set_kernel_modules.